### ISSUE TITLE
Compound types aren't importable from types; patch package_exports or docs?

### ISSUE THREAD

**Issue by kstrauser** [schematics/schematics#435]
Based on the contents of `schematics/types/__init__.py` and `docs/api/types.rst`, I expected to be able to run `from schematics.types import ModelType`. However, that doesn't work and I have to explicitly `from schematics.types.compound import ModelType`. Is this intentional or accidental?

If intentional: we should update `types.rst` to differentiate between types importable from `schematics.types` and those that require explicit import paths.

If accidental: we should update `schematics.util.package_exports` to correctly identify those models.

I'm leaning toward the latter. Any comments to the contrary before I proceed?

**Comment by bintoro** [comment]
> I expected to be able to run `from schematics.types import ModelType`. However, that doesn't work and I have to explicitly `from schematics.types.compound import ModelType`.

Historically it has been like this, but I've fixed it for v2.0 in 0bd471332c8860fc86cc314317941a604d9a44de.

**Comment by kstrauser** [comment]
Ahh, excellent! Even though I know my way around the codebase, I _try_ to assess the docs like I'm using Schematics for the first time.

**Comment by bintoro** [comment]
Yeah, and I think it totally makes sense to have all types importable from `schematics.types`. Why this hasn't been so in the past, I don't know, but I suspect it may be because of circular dependencies.

I've eliminated some circular dependencies previously, but this one could not be helped without resorting to monkey-patching: references to `Model` and `ModelMeta` within `types.compound` are inserted retroactively by `schematics/__init__.py`.
